Cython/Compiler/PyrexTypes.py

```diff
@@ -3098,6 +3098,45 @@ class CFuncType(CType):
 
     subtypes = ['return_type', 'args']
 
+    class ExceptionValue:
+        def __init__(self, python_value, c_repr, type):
+            self.python_value = python_value
+            self.c_repr = c_repr
+            self.type = type
+
+        def __eq__(self, other):
+            if not isinstance(other, CFuncType.ExceptionValue):
+                return NotImplemented
+            # only the python_value is used for equality comparison. This allows
+            # things like "-1 == -1.0" to be treated as the same function signature
+            return self.python_value == other.python_value
+
+        def __str__(self):
+            # repr is used in code-generation
+            return str(self.c_repr)
+
+        def may_be_nan(self):
+            if not self.type.is_float:
```

Comment:
What about `complex`, BTW, doesn't that suffer from the same problem?

Reply:
https://github.com/cython/cython/issues/6916 - there's quite a bit broken there